pkg/controller/controller_instance_test.go

```diff
@@ -440,6 +440,107 @@ func TestReconcileServiceInstanceResolvesReferences(t *testing.T) {
 	}
 }
 
+// TestReconcileServiceInstanceResolvesReferences tests a simple successful
+// reconciliation and making sure that the ServicePlanRef is correctly
+// resolved if the ServiceClassRef is already set.
+func TestReconcileServiceInstanceResolvesReferencesServiceClassRefAlreadySet(t *testing.T) {
```

Comment:
This is good, I added some smaller tests that only target resolveReferences method that make tests little easier to write / follow. Just a thought.